### PR TITLE
feat: go to Assignment tab after creation and warn if existing without assigning

### LINKED ISSUE #14848: [CAL-3606] Team events: go to Assignment tab after creation and warn if existing without assigning
For team event-types, after clicking "continue"

CleanShot 2024-05-01 at 19 27 03@2x

It should bring you directly to the **Assignment** tab instead of the **Event Setup** tab

CleanShot 2024-05-01 at 19 27 28@2x

If you try to exit out of an event type without assigning anyone, we should [throw a modal warning that you have not assigned anyone](https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Live?type=design&node-id=6279-62333&mode=design&t=jgaz5OvjQRqyaTRC-11). 
CleanShot 2024-05-03 at 10 23 00@2x

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3606](https://linear.app/calcom/issue/CAL-3606/team-events-include-assignment-in-event-type-creation-dialog)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

Fixes #14848 

Screenshot 2024-05-09 221817

### CODE REVIEW

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (supalarry) [inline apps/web/components/eventtype/EventTeamTab.tsx:3]: For reviewers: we are using next.js at version 13 and "next/router" has been deprecated, so we should import "useRouter" from "next/navigation", but "router.events" does not exist in version 13 so looks like need to roll with the deprecated import.

https://github.com/vercel/next.js/discussions/42016?sort=new

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (supalarry) [inline apps/web/components/eventtype/EventTeamTab.tsx:402]: `pending` in `setpendingRoute` should be written as `setPendingRoute`.

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (supalarry) [inline apps/web/components/eventtype/EventTeamTab.tsx:415]: Shall it be a more descriprive error - aborter rotue change to ${url} because noone was assigned to team event?

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (supalarry) [inline apps/web/components/eventtype/EventTeamTab.tsx:411]: Is there a reason to have "handleRouteChange" inside "useEffect" ? Why not move it out and wrap it in `useCallback` and then call it on "routeChangeStart"

File: apps/web/components/eventtype/AssignmentWarningDialog.tsx
Comment (supalarry) [inline apps/web/components/eventtype/AssignmentWarningDialog.tsx:8]: if it is an interface we don't need to prefix it with "I", so we can simply call it `AssignmentWarningDialogProps` or just `Props` since its straight forward that they are props for the component.

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (supalarry) [inline apps/web/components/eventtype/EventTeamTab.tsx:403]: Can we rename `isConfirm` to something else - it's not obvious that it is used by the "AssignmentWarningDialog". How about `leaveWithoutAssigningHosts` ?

**Review by supalarry — COMMENTED** [review]
Great job Anik! I confirm that functionality works as expected, but have some small refactor ideas below. Let me know what you think!

File: apps/web/public/static/locales/en/common.json
Comment (Udit-takkar) [inline apps/web/public/static/locales/en/common.json:2431]: move these keys above "ADD_NEW_STRINGS_ABOVE_THIS_LINE_TO_PREVENT_MERGE_CONFLICTS"

**Comment by anikdhabal** [comment]
@supalarry @Udit-takkar thanks for your reviews.

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (anikdhabal) [reply apps/web/components/eventtype/EventTeamTab.tsx:411]: There is no specific reason for it. Do we really need to cache the function since it only runs on route events and checks the assignee numbers?

**Review by supalarry — DISMISSED** [review]
Great work and thank you for the contribution! : )

**Comment by keithwillcode** [comment]
@ciaranha Would like your approval here before merging

**Comment by ciaranha** [comment]
> If you try to exit out of an event type without assigning anyone...

It should only throw this modal when you go to leave the entire event type, not a tab within the event type.

**Review by keithwillcode — DISMISSED** [review]
Marking as requested changes so we can address @ciaranha's comment about the tab versus page.

**Comment by anikdhabal** [comment]
@keithwillcode I've made the requested changes.

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
Uploading Screen Recording 2024-05-10 at 6.47.15 PM.mov…


You can navigate to other page without saving the hosts

**Comment by anikdhabal** [comment]
> Uploading Screen Recording 2024-05-10 at 6.47.15 PM.mov…
> 
> You can navigate to other page without saving the hosts

Thanks for the review. I've fixed that.

**Review by Udit-takkar — DISMISSED** [review]
I still sometimes don't see any modal when i navigate to any other page without assigning team members.

How to reproduce?
1) Create a collective team event type 
2) Go to event setup tab and change location and click save 
and then navigate to any other page by clicking on 'Bookings'

**Comment by anikdhabal** [comment]
> I still sometimes don't see any modal when i navigate to any other page without assigning team members.
> 
> How to reproduce?
> 
> 1. Create a collective team event type
> 2. Go to event setup tab and change location and click save
>    and then navigate to any other page by clicking on 'Bookings'

Hey @Udit-takkar why do we need show the modal when navigating to any other page from any other tab? We should only show the modal if someone navigates from the assignment tab without assigning anybody. Correct?

File: apps/web/modules/event-types/views/event-types-single-view.tsx
Comment (supalarry) [inline apps/web/modules/event-types/views/event-types-single-view.tsx:214]: If we want to show warning dialog only when navigating to a non `/event-types` route, do we need the `url === "/event-types"` check on line 214?

File: apps/web/components/eventtype/AssignmentWarningDialog.tsx
Comment (supalarry) [inline apps/web/components/eventtype/AssignmentWarningDialog.tsx:24]: You know what would be cool, if `Go Back and Assign` is clicked, that it re-directs to the `/event-types/:id?tabName=team` to land user on the page where he / she can assign the user. Otherwise the user needs to figure out which tab to press for assignement.

Screenshot 2024-05-17 at 10 15 04

Comment (anikdhabal) [reply apps/web/components/eventtype/AssignmentWarningDialog.tsx:24]: sure Lauris, that's a cool idea.

Comment (supalarry) [reply apps/web/components/eventtype/AssignmentWarningDialog.tsx:24]: Amazing! Thank you Anik!

File: apps/web/modules/event-types/views/event-types-single-view.tsx
Comment (anikdhabal) [reply apps/web/modules/event-types/views/event-types-single-view.tsx:214]: This only checks the back button, because the route is also /event-types.

**Comment by anikdhabal** [comment]
As we aborted the route change, we need to update the test. cc @Udit-takkar @supalarry

**Comment by supalarry** [comment]
> As we aborted the route change, we need to update the test. cc @Udit-takkar @supalarry

Sorry if i missed something:
1. What do you mean with "aborted the route change" ?
2. Which test?

**Comment by supalarry** [comment]
> > > As we aborted the route change, we need to update the test. cc @Udit-takkar @supalarry
> > 
> > 
> > Sorry if i missed something:
> > 
> > 1. What do you mean with "aborted the route change" ?
> > 2. Which test?
> 
> Aborted means we stop to redirect a specific route without confirmation, and that's why I think some end-to-end tests fail. What do you think? And Apart from this all things are updated and ready to approve.

Sometimes e2e tests fail for unrelated reasons, and they simply have to be re-run and then most of the time they become "green".

I merged main into this branch let's see if e2e fail, if so I will try re-running failed ones.

**Comment by anikdhabal** [comment]
> > > > As we aborted the route change, we need to update the test. cc @Udit-takkar @supalarry
> > > 
> > > 
> > > Sorry if i missed something:
> > > 
> > > 1. What do you mean with "aborted the route change" ?
> > > 2. Which test?
> > 
> > 
> > Aborted means we stop to redirect a specific route without confirmation, and that's why I think some end-to-end tests fail. What do you think? And Apart from this all things are updated and ready to approve.
> 
> Sometimes e2e tests fail for unrelated reasons, and they simply have to be re-run and then most of the time they become "green".
> 
> I merged main into this branch let's see if e2e fail, if so I will try re-running failed ones.

yeah that's true. Thanks Lauris

**Review by supalarry — DISMISSED** [review]

**Comment by supalarry** [comment]
All the checks are green and the read ones I think simply requires someone's approval - once someone with the deployment rights approves this PR I think the red ones will be actually triggered.

**Comment by anikdhabal** [comment]
I think I need to fix the end-to-end tests. These tests keep failing.

**Review by Udit-takkar — APPROVED** [review]
Adding to queue again

**Comment by shirazdole** [comment]
dude, thank you @anikdhabal  !! the amount of support tickets i get for this in particular is a lot. huge help for product experience for our customers ✨

**Pull request merged**
